kyo-kernel/shared/src/main/scala/kyo/kernel/Loop.scala

```diff
@@ -357,18 +357,18 @@ object Loop:
         inline run: Safepoint ?=> (A, B, C, D) => Outcome4[A, B, C, D, O] < S
     )(using inline _frame: Frame, safepoint: Safepoint): O < S =
         @nowarn("msg=anonymous")
-        @tailrec def loop(i1: A, i2: B, i3: C, i4: D)(v: Outcome4[A, B, C, D, O] < S = run(i1, i2, i3, i4))(using Safepoint): O < S =
+        @tailrec def loop(v: Outcome4[A, B, C, D, O] < S)(using Safepoint): O < S =
             v match
                 case next: Continue4[A, B, C, D] @unchecked =>
-                    loop(next._1, next._2, next._3, next._4)()
+                    loop(run(next._1, next._2, next._3, next._4))
                 case kyo: KyoSuspend[IX, OX, EX, Any, Outcome4[A, B, C, D, O], S] @unchecked =>
                     new KyoContinue[IX, OX, EX, Any, O, S](kyo):
                         def frame = _frame
                         def apply(v: OX[Any], context: Context)(using Safepoint) =
-                            loop(i1, i2, i3, i4)(kyo(v, context))
```

Comment:
The change also reduces the memory footprint. For example, here it avoids four fields in the `KyoSuspend` closure since the inputs don't need to be captured anymore.